tests/raftstore/cluster.rs

```diff
@@ -513,49 +500,39 @@ impl<T: Simulator> Cluster<T> {
     }
 
     // NOTE: if you have set transport hooks before, call this function will overwrite them
-    pub fn partition(&mut self, s1: Arc<HashSet<u64>>, s2: Arc<HashSet<u64>>) {
-        for node_id in s1.iter() {
+    pub fn partition(&self, s1: Vec<u64>, s2: Vec<u64>) {
+        for node_id in &s1 {
             let filter = new_partition_filter(s2.clone());
-            self.sim.rl().hook_transport(*node_id, vec![RwLock::new(filter)]);
+            self.sim.rl().hook_transport(*node_id, vec![filter]);
         }
-        for node_id in s2.iter() {
+        for node_id in &s2 {
             let filter = new_partition_filter(s1.clone());
-            self.sim.wl().hook_transport(*node_id, vec![RwLock::new(filter)]);
+            self.sim.wl().hook_transport(*node_id, vec![filter]);
         }
     }
 
     pub fn reset_transport_hooks(&mut self) {
-        let sim = &self.sim.rl();
+        let sim = self.sim.wl();
         for node_id in sim.get_node_ids() {
             sim.hook_transport(node_id, vec![]);
         }
     }
-}
 
-struct PartitionFilter {
-    node_ids: Arc<HashSet<u64>>,
-    drop: bool,
-}
-
-impl Filter for PartitionFilter {
-    fn before(&mut self, msg: &RaftMessage) -> bool {
-        self.drop = self.node_ids.contains(&msg.get_to_peer().get_store_id());
-        self.drop
-    }
-    fn after(&mut self, r: Result<()>) -> Result<()> {
-        if self.drop {
-            return Err(Error::Timeout("drop by PartitionPacket in SimulateTransport".to_string()));
+    pub fn hook_transport_drop_packet(&self, rate: u32) {
```

Comment:
because we haven't expose a similar API like below for Cluster now

    fn hook_transport(&self, node_id: u64, filters: Vec<Box<Filter>>);

and we also need expose an API for Cluster if we want to add filters outerside.

    fn get_node_ids()

and we still need provide filter's code like `hook_transport_drop_packet` outerside Cluster

that's not a big deal, so I leave it like this now...

if I find we need more and more things in Cluster, I should change it.